proto/rill/runtime/v1/api.proto

```diff
@@ -497,6 +504,19 @@ message GenerateMetricsViewFileResponse {
   bool ai_succeeded = 1;
 }
 
+message GenerateChartSpecRequest {
+  string instance_id = 1 [(validate.rules).string = {pattern: "^[_\\-a-zA-Z0-9]+$"}];
+  string chart = 2;
+  string table = 3;
+  string metrics_view = 4;
```

Comment:
Since only one of these should be set, use a `oneof`.

But it seems weird to be able to pass a `chart` here, since it's kind of self-referential. And with Vega spec validation, it won't even be possible to create a chart without already having a Vega spec set in it?

Also, how are you planning on adding metrics view support?

I'm wondering if it would make more sense to pass `resolver` and `resolver_properties` here, since that's what the chart will be based on. And then having a separate AI endpoint for generating the resolver from e.g. a table name (generate plain SQL resolver) or a metrics view (generate metrics SQL)